src/uu/hashsum/src/hashsum.rs

```diff
@@ -531,11 +540,32 @@ where
                     Err(e) => return Err(e.map_err_context(|| "failed to read file".to_string())),
                 };
                 let (ck_filename, sum, binary_check) = match gnu_re.captures(&line) {
-                    Some(caps) => (
-                        caps.name("fileName").unwrap().as_str(),
-                        caps.name("digest").unwrap().as_str().to_ascii_lowercase(),
-                        caps.name("binary").unwrap().as_str() == "*",
-                    ),
+                    Some(caps) => {
+                        // We set the format during first pass, as mixing of formats is disallowed.
+                        if bsd_reversed.is_none() {
+                            let is_bsd_reversed = caps.name("binary").is_none();
+                            let format_marker = if is_bsd_reversed {
+                                ""
+                            } else {
+                                r"(?P<binary>[ \*])"
+                            }
+                            .to_string();
+
+                            bsd_reversed = Some(is_bsd_reversed);
+                            gnu_re = gnu_re_template(&bytes_marker, &format_marker)?;
+                        }
+
+                        // Default mode for Reversed BSD format is 'text'
+                        (
+                            caps.name("fileName").unwrap().as_str(),
```

Comment:
any reason you aren't managing the error here?

Reply:
You mean possible unwrap error? The capture group is non-optional, so it's impossible it'd match this arm without that value being present.